pawnShop/src/main/java/com/example/pawnShop/Service/ProductTypeServiceImp.java

```diff
@@ -0,0 +1,82 @@
+package com.example.pawnShop.Service;
+
+import com.example.pawnShop.Dto.Product.ProductTypeDto;
+import com.example.pawnShop.Factory.Contract.ProductFactory;
+import com.example.pawnShop.ManualMapper.Contract.ProductManualMapper;
+import com.example.pawnShop.Repository.ProductTypeRepository;
+import com.example.pawnShop.Service.Contract.ProductTypeService;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Transactional
+@Service
+@RequiredArgsConstructor
+public class ProductTypeServiceImp implements ProductTypeService {
+
+    private final ProductTypeRepository repository;
+    private final ProductManualMapper mapper;
+    private final ProductFactory factory;
+
+    @Override
+    public ProductTypeDto getProductTypeById(UUID id) {
+
+        //todo decide how to manage exceptions
+        var productType = repository.findById(id).orElseThrow();
+
+        return mapper.mapToProductTypeDto(productType);
+    }
+
+    @Override
+    public List<ProductTypeDto> getAllProductsTypes() {
+        return repository.findAll().stream().map(mapper::mapToProductTypeDto).toList();
+    }
+
+    @Override
+    public String addProductType(String name) {
+        if(name.isBlank() || name.isEmpty()){
+            return "";
+        }
+        if(!isWordApproved(name)){
+            return "";
+        }
+        var newProduct = factory.createProductType(name);
+
+        repository.save(newProduct);
+
+        return name;
+    }
+
+    @Override
```

Comment:
Add @Transactional to all Crud operations 
you can read more about it here:https://docs.spring.io/spring-framework/reference/data-access/transaction/declarative/annotations.html

As a summary - @Transactional = save everything if successful or save nothing if not ;)